src/pkgTypes.ts

```diff
@@ -0,0 +1,97 @@
+import type { RawPkg, StyleType } from './@types/pkg';
+import type { File } from './jsDelivr';
+import { datadog } from './utils/datadog';
+
+const styleFileExtensions: StyleType[] = ['css', 'less', 'scss'];
+const jsFileExtensions = ['js', 'mjs', 'cjs'];
+
+function createFilePattern(extensions: string[]): RegExp {
+  const extPattern = extensions.join('|');
+  return new RegExp(
+    `^(?:(?!\\/(docs?|documentation|examples?|samples?|demos?|tests?)\\/)(?!\\/[._]).)+\\.(${extPattern})$`
```

Comment:
Can be nice to provide a regex101 link to see what you are looking to exclude/include in this; negative look ahead/behind are hard to understand :p

Reply:
https://github.com/algolia/npm-search/pull/683/commits/8ff01012a86471b8ba3a013efd7c8731f1557ee8